core/arch/arm/plat-ti/platform_config.h

```diff
@@ -153,9 +153,18 @@
 
 #endif /* CFG_WITH_PAGER */
 
+#if defined(CFG_SECURE_DATA_PATH)
+/* Locate SDP memory at the end of TZDRAM */
+#define CFG_TEE_SDP_MEM_BASE	(TZDRAM_BASE + \
+				TZDRAM_SIZE - \
+				CFG_TEE_SDP_MEM_SIZE)
+#endif
+
 #define CFG_TA_RAM_START	ROUNDUP((TZDRAM_BASE + CFG_TEE_RAM_VA_SIZE), \
 					CORE_MMU_DEVICE_SIZE)
-#define CFG_TA_RAM_SIZE		ROUNDDOWN((TZDRAM_SIZE - CFG_TEE_RAM_VA_SIZE), \
+
+#define CFG_TA_RAM_SIZE		ROUNDDOWN((TZDRAM_SIZE - CFG_TEE_RAM_VA_SIZE) - \
+					  CFG_TEE_SDP_MEM_SIZE, \
```

Comment:
Sorry I missed something in my previous comment. You probably want `CFG_TEE_SDP_MEM_SIZE` set to `0x0` when `CFG_SECURE_DATA_PATH != y`? If so, `plat-ti/conf.mk` should contain:

```
ifeq ($(CFG_SECURE_DATA_PATH),y)
CFG_TEE_SDP_MEM_SIZE ?= 0x00400000
else
CFG_TEE_SDP_MEM_SIZE ?= 0x0
endif
```

Anyway:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

Reply:
I like that, updated.